foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -880,44 +946,50 @@ def retry_lost_survey_jobs() -> None:
     nomad_host = get_env_variable("NOMAD_HOST")
     nomad_port = get_env_variable("NOMAD_PORT", "4646")
     nomad_client = Nomad(nomad_host, port=int(nomad_port), timeout=30)
-    lost_jobs = []
 
-    for job in potentially_lost_jobs:
-        try:
-            # Surveyor jobs didn't always have nomad_job_ids. If they
-            # don't have one then by this point they've definitely died.
-            if job.nomad_job_id:
-                job_status = nomad_client.job.get_job(job.nomad_job_id)["Status"]
-            else:
-                job_status = "absent"
-
-            # If the job is still pending, then it makes sense that it
-            # hasn't started and if it's running then it may not have
-            # been able to mark the job record as started yet.
-            if job_status != "pending" and job_status != "running":
-                logger.debug(("Determined that a survey job needs to be requeued because its"
-                             " Nomad Job's status is: %s."),
-                            job_status,
-                            job_id=job.id
+    if potentially_lost_jobs.count() == 0:
+        return
+
+    paginator = Paginator(potentially_lost_jobs, 200)
+    for page_idx in range(1, paginator.num_pages + 1):
```

Comment:
This looks funny because pages are 0 indexed and range isn't inclusive of the final value, but I think it's right. Maybe needs a comment? Probably not.